apps/services/user-notification/src/app/modules/notifications/notifications.service.ts

```diff
@@ -59,18 +60,108 @@ export class NotificationsService {
     private readonly notificationModel: typeof Notification,
   ) {}
 
-  private async formatAndMapNotification(
+  async performGraphQLRequest(query: string) {
+    try {
+      const response = await axios.post(
+        CONTENTFUL_GQL_ENDPOINT,
+        { query },
+        {
+          headers: {
+            'content-type': 'application/json',
+            authorization: `Bearer ${ACCESS_TOKEN}`,
+          },
+        },
+      )
+      return response.data
+    } catch (error) {
+      if (error.response) {
+        throw new BadRequestException(error.response.data)
+      } else {
+        this.logger.error('GraphQL Request Failed', { error })
+        throw new InternalServerErrorException('Internal Server Error')
+      }
+    }
+  }
+
+  async getSenderOrganization(
+    senderId: string,
+    locale?: Locale,
+  ): Promise<SenderOrganization> {
+    locale = mapToLocale(locale as Locale)
+    const cacheKey = `org-${senderId}-${locale}`
+    const cachedOrganization = await this.cacheManager.get<SenderOrganization>(
+      cacheKey,
+    )
+    if (cachedOrganization) {
+      this.logger.info(`Cache HIT for: ${cacheKey}`, cachedOrganization)
+      return cachedOrganization
+    } else {
+      this.logger.warn(`Cache MISS for: ${cacheKey}`)
+    }
+    const contentfulOrganizationQuery = `{
+      organizationCollection(where: {kennitala: "${senderId}"}, locale: "${mapToContentfulLocale(
+      locale,
+    )}") {
+        items {
+          title
+        }
+      }
+    }`
+    const res = await this.performGraphQLRequest(contentfulOrganizationQuery)
+    const organizationTitle =
+      res.data.organizationCollection.items[0]?.title ?? undefined
```

Comment:
Is items guaranteed to be at least an empty array?
Otherwise perhaps the optional array accessor `?.[0]`:
```suggestion
      res.data.organizationCollection.items?.[0]?.title ?? undefined
```

Reply:
@coderabbit what do you think?

Reply:
good call 👍